package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -660,58 +616,87 @@ cdef class DCDFile:
         start = start if not start is None else 0
         step = step if not step is None else 1
 
+        # c delcs
+        cdef int stop_ = stop
+        cdef int start_ = start
+        cdef int step_ = step 
 
         cdef int n
         n = len(range(start, stop, step))
-
+        cdef int natoms
         cdef np.ndarray[np.int64_t, ndim=1] c_indices
         if indices is None:
-            c_indices = np.arange(self.natoms, dtype=np.int64)
+            c_indices = np.PyArray_Arange(0, self.natoms, 1, np.NPY_INT64)
             natoms = self.natoms
         else:
             natoms = len(indices)
             c_indices = np.asarray(indices, dtype=np.int64)
-
+        
+        cdef np.npy_intp[3] dims
         cdef int hash_order = -1
         if order == 'fac':
-            shape = (n, natoms, self.ndims)
+            dims[0] = n
+            dims[1] = natoms
+            dims[2] = self.ndims
             hash_order = 1
         elif order == 'fca':
-            shape = (n, self.ndims, natoms)
+            dims[0] = n
+            dims[1] = self.ndims
+            dims[2] = natoms
             hash_order = 2
         elif order == 'afc':
-            shape = (natoms, n, self.ndims)
+            dims[0] = natoms
+            dims[1] = n
+            dims[2] = self.ndims
             hash_order = 3
         elif order == 'acf':
-            shape = (natoms, self.ndims, n)
+            dims[0] = natoms
+            dims[1] = self.ndims
+            dims[2] = n
             hash_order = 4
         elif order == 'caf':
-            shape = (self.ndims, natoms, n)
+            dims[0] = self.ndims
+            dims[1] = natoms
+            dims[2] = n
             hash_order = 5
         elif order == 'cfa':
+            dims[0] = self.ndims
+            dims[1] = n
+            dims[2] = natoms
             hash_order = 6
-            shape = (self.ndims, n, natoms)
         else:
             raise ValueError("unkown order '{}'".format(order))
 
-        cdef np.ndarray[FLOAT_T, ndim=3] xyz = np.empty(shape, dtype=FLOAT)
-        cdef np.ndarray[DOUBLE_T, ndim=2] box = np.empty((n, 6))
-
-
-        cdef np.ndarray xyz_tmp = np.empty((self.natoms, self.ndims), dtype=FLOAT, order='F')
-        cdef int ok, i
-
-        if start == 0 and step == 1 and stop == self.n_frames:
+        cdef np.ndarray[FLOAT_T, ndim=3] xyz = np.PyArray_EMPTY(3, dims, np.NPY_FLOAT32, 0)
```

Comment:
can reuse the scratch buffer here?

Reply:
If we are deprecating, I might just leave it alone for now. Main reading methods use buffer now.